mptt/models.py

```diff
@@ -182,6 +182,11 @@ def insertion_target_filters(self, instance, order_insertion_by):
                 field = instance._meta.get_field(field_name)
                 value = field.pre_save(instance, True)
 
+            # not really sure how this works, but fixes #340 
+            if value is None:
+                value = bool(filter_suffix=='__lt')
```

Comment:
`==` already results in a boolean, so the `bool()` here is unnecessary. Also, pep8 has spaces either side of the `==`.

I think nulls sort first (in all databases? not sure - can you check?). So probably what this is doing is finding this index:

`null, null, <<insert new null node here>>, 'A', 'A1'`
`'A1', 'A', <<insert new null node here>>, null, null`

i.e.

* if sorting the attribute in ascending order, find the index of the first node that's _not_ null, and insert there.
* if sorting the attribute in ascending order, find the index of the first node that _is_ null, and insert there.

need some more comments, and probably a test for both sort cases I think.